packages/react-strict-dom/src/native/modules/useStrictDOMElement.js

```diff
@@ -38,68 +40,89 @@ function proxy(nativeRef: React.RefObject<Node>, name: string) {
 export function useStrictDOMElement<T>(
   ref: React.RefSetter<Node>,
   { tagName }: Options
-): React.RefObject<T | null> {
-  const nativeRef = React.useRef<T | null>(null);
+): CallbackRef<T> {
   const { scale: viewportScale } = useViewportScale();
 
-  React.useImperativeHandle(ref, () => {
-    return {
-      get __nativeTag() {
-        const node = nativeRef.current as Node;
-        return node?.__nativeTag;
-      },
-      addEventListener: proxy(nativeRef, 'addEventListener'),
-      animate: proxy(nativeRef, 'animate'),
-      blur: proxy(nativeRef, 'blur'),
-      click: proxy(nativeRef, 'click'),
-      get complete() {
-        if (tagName === 'img') {
-          const node = nativeRef.current as Node;
-          if (node?.complete == null) {
-            // Assume images are never pre-loaded in React Native
-            return false;
+  const elementCallback = useElementCallback(
+    React.useCallback(
+      // $FlowFixMe[unclear-type]
+      (node: any) => {
+        const obj = {
```

Comment:
You should memoize this `obj`.

There are multiple scenarios in which the same instance is passed to multiple refs:

- An enclosing `Activity` becomes hidden, then visible. (DOM nodes do not get deallocated when `Activity` becomes hidden, so when `Activity` becomes visible, that same DOM node would be reattached to refs.)
- A new ref is supplied to a native component that is already mounted. (The old ref and the new ref would receive instances that are `===`.)